packages/open-next/src/overrides/wrappers/cloudflare-node.ts

```diff
@@ -41,15 +41,16 @@ const handler: WrapperHandler<InternalEvent, InternalResult> =
       }): Writable {
         const { statusCode, cookies, headers } = prelude;
 
-        const responseHeaders = new Headers(headers);
-        for (const cookie of cookies) {
-          responseHeaders.append("Set-Cookie", cookie);
-        }
-
         // TODO(vicb): this is a workaround to make PPR work with `wrangler dev`
         // See https://github.com/cloudflare/workers-sdk/issues/8004
         if (url.hostname === "localhost") {
-          responseHeaders.set("Content-Encoding", "identity");
+          headers["content-encoding"] = "identity";
```

Comment:
this now mutates the headers object coming from outside the function, which changes its behavior (it now has a side effect on the caller values).